### PR TITLE
Initial rework pkg/registry to use `begin` pattern

### PR DESCRIPTION
**Pull request by denis-tingaikin**
Signed-off-by: Denis Tingaikin <denis.tingajkin@xored.com>

## Description
The main motivation of this PR is making healing for `registry` as simple as we have for `networkservice`. 
All retry and heal logic has been moved into clientside.

To achieve this functionality has been reworked all overcomplicated chain elements like `connect`, `heal`, `expire`, `refresh` to use `begin` pattern.

See at `begin` pattern https://github.com/networkservicemesh/sdk/pull/1072 for `networkservice`.
 

## Issue link
Closes https://github.com/networkservicemesh/sdk/issues/1170
Closes https://github.com/networkservicemesh/deployments-k8s/issues/4247
Closes https://github.com/networkservicemesh/deployments-k8s/issues/3938
Closes https://github.com/networkservicemesh/deployments-k8s/issues/1017

## How Has This Been Tested?

- [X] Added unit testing to cover
- [X] Tested manually
- [ ] Tested by integration testing
- [ ] Have not tested

## Types of changes

- [X] Bug fix
- [X] New functionallity
- [ ] Documentation
- [X] Refactoring
- [ ] CI

### CODE REVIEW

**Comment by denis-tingaikin** [comment]
I also have some local changes for https://github.com/networkservicemesh/sdk/issues/1196 for this patch.

Will push them soon.


Please do not merge yet.

File: pkg/registry/common/begin/close_server_test.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package begin_test
+
+// func TestCloseServer(t *testing.T) {
```
Comment (glazychev-art) [inline pkg/registry/common/begin/close_server_test.go:19]: Do we need these tests?

File: pkg/networkservice/chains/nsmgr/unix_test.go [hunk 2]
```diff
@@ -41,7 +41,7 @@ import (
 func Test_Local_NoURLUsecase(t *testing.T) {
 	t.Cleanup(func() { goleak.VerifyNone(t) })
 
-	ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
+	ctx, cancel := context.WithTimeout(context.Background(), time.Hour*5)
```
[the remark this answers is not in the document]
Comment (denis-tingaikin) [reply pkg/networkservice/chains/nsmgr/unix_test.go]: Thanks, done.

File: pkg/registry/common/begin/close_server_test.go (hunk 1 above)
Comment (denis-tingaikin) [reply pkg/registry/common/begin/close_server_test.go]: Thanks, done.

File: pkg/networkservice/chains/nsmgr/single_test.go [hunk 3]
```diff
@@ -167,7 +167,7 @@ func Test_ShouldParseNetworkServiceLabelsTemplate(t *testing.T) {
 func Test_UsecasePoint2MultiPoint(t *testing.T) {
 	t.Cleanup(func() { goleak.VerifyNone(t) })
 
-	ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute*5)
```
[the remark this answers is not in the document]
Comment (denis-tingaikin) [reply pkg/networkservice/chains/nsmgr/single_test.go]: Thanks, done.

File: pkg/registry/common/recvfd/server_test.go [hunk 4]
```diff
@@ -71,15 +74,15 @@ func TestNseRecvfdServerClosesFile(t *testing.T) {
 	var ctx, cancel = context.WithTimeout(context.Background(), 1*time.Second)
 	defer cancel()
 
-	var nsRegistry = registrychain.NewNetworkServiceRegistryServer(
-		registryserialize.NewNetworkServiceRegistryServer(),
+	var nsRegistry = chain.NewNetworkServiceRegistryServer(
+		// begin.NewNetworkServiceRegistryServer(),
```
Comment (NikitaSkrynnik) [inline pkg/registry/common/recvfd/server_test.go:78]: Is this correct?

**Pull request merged**